lib/AACStream/aacs.js

```diff
@@ -0,0 +1,120 @@
+/**
+ * mux.js
+ *
+ * Copyright (c) 2016 Brightcove
+ * All rights reserved.
+ *
+ * A stream-based aac to mp4 converter. This utility can be used to
+ * deliver mp4s to a SourceBuffer on platforms that support native
+ * Media Source Extensions.
+ */
+'use strict';
+var Stream = require('../utils/stream.js');
+
+var
+
+/**
+ * Splits an incoming stream of binary data into ADTS and ID3 Frames.
+ */
+
+ AacStream = function() {
+  var
+    everything,
+    receivedTimeStamp = false,
+    timeStamp = 0;
+
+  AacStream.prototype.init.call(this);
+
+  this.setTimestamp = function (frame) {
```

Comment:
The thing is that it is taking the value that is being set. It receives an ID3 tag that I then use to set the `timestamp` variable. Then whenever we emit an ADTS frame we check the timestamp value and add in the necessary info. It is setup like this because we can have an arbitrary amount of ADTS frames following an ID3 tag. I think we should discuss alternatives in person.